### ISSUE TITLE
Dropout issue on Spark

### ISSUE THREAD

**Issue by sandgrules** [deeplearning4j/deeplearning4j#7012]
#### Issue Description

I have a model that I'm training over a Spark cluster. The model trains fine without dropout layers, but as soon as I add a dropout layer I get an error:

Exception in thread "main" org.apache.spark.SparkException: Job aborted due to stage failure: Task 54 in stage 7.0 failed 4 times, most recent failure: Lost task 54.3 in stage 7.0 (TID 415, data11, executor 16): java.lang.IllegalStateException: Cannot perform backprop: Dropout mask array is absent (already cleared?)
        at org.nd4j.base.Preconditions.throwStateEx(Preconditions.java:626)                           
        at org.nd4j.base.Preconditions.checkState(Preconditions.java:253)                             
        at org.deeplearning4j.nn.conf.dropout.Dropout.backprop(Dropout.java:154)                      
        at org.deeplearning4j.nn.layers.AbstractLayer.backpropDropOutIfPresent(AbstractLayer.java:309)
        at org.deeplearning4j.nn.layers.BaseLayer.backpropGradient(BaseLayer.java:106)         
        at org.deeplearning4j.nn.graph.vertex.impl.LayerVertex.doBackward(LayerVertex.java:148)
        at org.deeplearning4j.nn.graph.ComputationGraph.calcBackpropGradients(ComputationGraph.java:2592)
        at org.deeplearning4j.nn.graph.ComputationGraph.computeGradientAndScore(ComputationGraph.java:1366)
        at org.deeplearning4j.nn.graph.ComputationGraph.computeGradientAndScore(ComputationGraph.java:1326)
        at org.deeplearning4j.optimize.solvers.BaseOptimizer.gradientAndScore(BaseOptimizer.java:160)
        at org.deeplearning4j.optimize.solvers.StochasticGradientDescent.optimize(StochasticGradientDescent.java:51)
        at org.deeplearning4j.optimize.Solver.optimize(Solver.java:52)
        at org.deeplearning4j.nn.graph.ComputationGraph.fitHelper(ComputationGraph.java:1149)
        at org.deeplearning4j.nn.graph.ComputationGraph.fit(ComputationGraph.java:1098)
        at org.deeplearning4j.nn.graph.ComputationGraph.fit(ComputationGraph.java:1006)
        at org.deeplearning4j.spark.impl.paramavg.ParameterAveragingTrainingWorker.processMinibatch(ParameterAveragingTrainingWorker.java:232)
        at org.deeplearning4j.spark.impl.paramavg.ParameterAveragingTrainingWorker.processMinibatchWithStats(ParameterAveragingTrainingWorker.java:274)
        at org.deeplearning4j.spark.api.worker.ExecuteWorkerMultiDataSetFlatMapAdapter.call(ExecuteWorkerMultiDataSetFlatMap.java:112)
        at org.deeplearning4j.spark.api.worker.ExecuteWorkerMultiDataSetFlatMapAdapter.call(ExecuteWorkerMultiDataSetFlatMap.java:56)
        at org.deeplearning4j.spark.api.worker.ExecuteWorkerPathMDSFlatMapAdapter.call(ExecuteWorkerPathMDSFlatMap.java:91)
        at org.deeplearning4j.spark.api.worker.ExecuteWorkerPathMDSFlatMapAdapter.call(ExecuteWorkerPathMDSFlatMap.java:55)
        at org.datavec.spark.transform.BaseFlatMapFunctionAdaptee.call(BaseFlatMapFunctionAdaptee.java:40)

The entire model is fairly large and so I haven't included it, but the code snippet where I'm including dropout: 
private def getSuperLayer(nIn: Int, nOut: Int): DenseLayer = {
    new DenseLayer.Builder().nIn(nIn).nOut(nOut).l1(l1).activation(Activation.LEAKYRELU).dropOut(.75).build()
  }

#### Version Information
Snippets from POM: 
        <dl4j.version>1.0.0-beta2</dl4j.version>
        <dl4j.spark.version>1.0.0-beta2_spark_2</dl4j.spark.version>
        <datavec.spark.version>1.0.0-beta2_spark_2</datavec.spark.version>
        <nd4j.backend>nd4j-native-platform</nd4j.backend>
Also using Scala 2.11.x

Has anyone else seen this error, or am I making an id - 10 - T error?

**Comment by maxpumperla** [comment]
@sandgrules thanks for reporting. we've had a lot of Spark related fixes recently. before we do anything, would you mind upgrading to 1.0.0-beta3 or even 1.0.0-SNAPSHOT to see if that works? I do vaguely remember we had dropout fixes.

**Comment by AlexDBlack** [comment]
Thanks for reporting, I'm closing this as a duplicate of this issue: https://github.com/deeplearning4j/deeplearning4j/issues/6841
The issue has already been fixed on master, and the fix is available on snapshots: https://deeplearning4j.org/docs/latest/deeplearning4j-config-snapshots
